config/src/main/java/com/alibaba/nacos/config/server/service/repository/extrnal/ExternalStoragePersistServiceImpl.java

```diff
@@ -436,9 +457,12 @@ public void insertOrUpdate(String srcIp, String srcUser, ConfigInfo configInfo,
     public void insertOrUpdate(String srcIp, String srcUser, ConfigInfo configInfo, Timestamp time,
             Map<String, Object> configAdvanceInfo, boolean notify) {
         try {
+            int updateCount = updateConfigInfo(configInfo, srcIp, srcUser, time, configAdvanceInfo, notify);
+            if (updateCount <= 0) {
+                addConfigInfo(srcIp, srcUser, configInfo, time, configAdvanceInfo, notify);
+            }
+        } catch (EmptyResultDataAccessException ex) { // No data, insert
```

Comment:
is there any cases that will throw EmptyResultDataAccessException in an update & add operation? same whit above.